### PR TITLE
Update Zinc and Iron API docs

### LINKED ISSUE #55: OpenCMISS-Iron API documentation is out-of-date
The OpenCMISS-Iron API documentation hasn't been updated since December 2016. This is caused by the migration of the OpenCMISS build system from Buildbot/make to Jenkins/CMake.
After discussion at last week's OpenCMISS meeting, it was decided that we will generate the documentation as part of the website build process.

### PR DESCRIPTION
**Pull request by inkybutton**
Fixes #55 and #57. 
@chrispbradley and @rchristie, please check and let me know if you're happy with it.

### CODE REVIEW

File: apidoc/Makefile [hunk 1]
```diff
@@ -1,11 +1,35 @@
-html: clean
-	cp -r zinc .working
-	cp LibZincDoxygen/api_2_api++_doc.py .working
-	cp -r LibZincDoxygen/libzinc_doxgen_script .working
-	cd .working; python api_2_api++_doc.py
-	cd .working/libzinc_doxgen_script; doxygen cpp_comments.doxygen
+html: clean zincapi ironapi
+
+zincapi:
+# Copies the OpenCMISS-Iron source code to a working directory then run a custom script, and then doxygen to generate the relevant API documentation.
```
Comment (rchristie) [inline apidoc/Makefile]: This comment mentions Iron not Zinc. It's worth updating both the comments for Zinc and Iron to briefly state what they actually do; Zinc runs a script to transfer doxygen from its C to C++ API, then generates the C++ API docs. Iron has a few different steps.

Comment (inkybutton) [reply apidoc/Makefile]: Hi @rchristie , thanks I have updated the comments to better describe what's going on. Let me know if you're happy with that.

**Comment by inkybutton** [comment]
Hi @chrispbradley , I have been OK'd by @rchristie regarding this merge. Do you have other comments you would like to add?

**Comment by chrispbradley** [comment]
Looks good to me. I'll merge.

**Pull request merged**